imageio/plugins/spe.py

```diff
@@ -575,172 +433,426 @@ def _open(self, char_encoding="latin1", check_filesize=True, sdt_meta=True):
+
+        nr = m.pop("NumROI", None)
+        nr = 1 if nr < 1 else nr
+        m["ROIs"] = roi_array_to_dict(m["ROIs"][:nr])
+
+        # chip sizes
+        m["chip_size"] = [m.pop(k, None) for k in ("xDimDet", "yDimDet")]
+        m["virt_chip_size"] = [m.pop(k, None) for k in ("VChipXdim", "VChipYdim")]
+        m["pre_pixels"] = [m.pop(k, None) for k in ("XPrePixels", "YPrePixels")]
+        m["post_pixels"] = [m.pop(k, None) for k in ("XPostPixels", "YPostPixels")]
+
+        # convert comments from numpy.str_ to str
+        m["comments"] = [str(c) for c in m["comments"]]
+
+        # geometric operations
+        g = []
+        f = m.pop("geometric", 0)
+        if f & 1:
+            g.append("rotate")
+        if f & 2:
+            g.append("reverse")
+        if f & 4:
+            g.append("flip")
+        m["geometric"] = g
+
+        # Make some additional information more human-readable
+        t = m["type"]
+        if 1 <= t <= len(Spec.controllers):
+            m["type"] = Spec.controllers[t - 1]
+        else:
+            m["type"] = None
+        r = m["readout_mode"]
+        if 1 <= r <= len(Spec.readout_modes):
+            m["readout_mode"] = Spec.readout_modes[r - 1]
+        else:
+            m["readout_mode"] = None
+
+        # bools
+        for k in (
+            "absorb_live",
+            "can_do_virtual_chip",
+            "threshold_min_live",
+            "threshold_max_live",
+        ):
+            m[k] = bool(m[k])
+
+        # Extract SDT-control metadata if desired
+        if sdt_control:
+            SDTControlSpec.extract_metadata(m, char_encoding)
+
+        return m
+
+    def _metadata_post_v3(self) -> Dict[str, Any]:
+        """Extract XML metadata from SPE v3 files
+
+        Returns
+        -------
+        dict with key `"__xml"`, whose value is the XML metadata
+        """
+
+        info = self._parse_header(Spec.basic, "latin1")
+        self._file.seek(info["xml_footer_offset"])
+        xml = self._file.read()
+        return {"__xml": xml}
+
+    def properties(self, index: int = ...) -> ImageProperties:
+        """Standardized ndimage metadata
+
+        Parameters
+        ----------
+        index
+            If the index is an integer, select the index-th frame and return
+            its properties. If index is an ellipsis (...), return the
+            properties of all frames in the file stacked along a new batch
+            dimension.
+
+        Returns
+        -------
+        properties
+            A dataclass filled with standardized image metadata.
+        """
+
+        if index is Ellipsis:
+            return ImageProperties(
+                shape=(self._len, *self._shape),
+                dtype=self._dtype,
+                n_images=self._len,
+                is_batch=True,
+            )
+        return ImageProperties(shape=self._shape, dtype=self._dtype, is_batch=False)
+
+    def _parse_header(
```

Comment:
This function, too, just appears to have moved. Could you confirm?

Reply:
This is new. The old plugin had no `properties()`.

Reply:
Sorry, you probably meant `_parse_header()`. The only change is that `char_encoding` is now a parameter. Previously, `self._char_encoding`  was used.

Reply:
Yes, I was referring to `char_encoding`. Somehow GH decided to refer to the past 4 lines instead of just referencing the definition that I selected ._. sorry for the confusion.